libdnf/utils/utils.cpp

```diff
@@ -172,41 +172,48 @@ bool haveFilesSameContent(const char * filePath1, const char * filePath2)
     return ret;
 }
 
+static bool
+saveFile(const char * filePath, const char * newFileContent, unsigned long newFileContentLen)
+{
+    int fd = -1;
+    if ((fd = open(filePath,  O_WRONLY | O_CREAT | O_TRUNC, 0644)) == -1)
+        return false;
+    lseek(fd, 0, SEEK_SET);
+    auto writtenSize = write(fd, newFileContent, newFileContentLen);
+    close(fd);
+    return (static_cast<unsigned long>(writtenSize) == newFileContentLen);
+}
+
 bool updateFile(const char * filePath, const char * newFileContent)
 {
     static constexpr int BLOCK_SIZE = 4096;
     bool ret = false;
-    int fd1 = -1;
+    int fd = -1;
     const char * tmpFileContent = newFileContent;
     auto newFileContentLen = strlen(newFileContent);
+
+    if ((fd = open(filePath,  O_RDONLY)) == -1)
+        return saveFile(filePath, newFileContent, newFileContentLen);
+    auto len1 = lseek(fd, 0, SEEK_END);
+    ret = true;
+    if (len1 == 0) {
+        close(fd);
+        return saveFile(filePath, newFileContent, newFileContentLen);
+    }
+    lseek(fd, 0, SEEK_SET);
+    char buf1[BLOCK_SIZE];
+    ssize_t readed;
     do {
-        if ((fd1 = open(filePath,  O_RDWR | O_CREAT, 0755)) == -1)
-            break;
-        auto len1 = lseek(fd1, 0, SEEK_END);
-        ret = true;
-        if (len1 == 0) {
-            lseek(fd1, 0, SEEK_SET);
-            auto writtenSize = write(fd1, newFileContent, newFileContentLen);
-            ret = (static_cast<unsigned long>(writtenSize) == newFileContentLen);
-            break;
+        readed = read(fd, buf1, BLOCK_SIZE);
+        if (memcmp(buf1, tmpFileContent, readed) != 0) {
+            close(fd);
+            return saveFile(filePath, newFileContent, newFileContentLen);
         }
-        lseek(fd1, 0, SEEK_SET);
-        char buf1[BLOCK_SIZE];
-        ssize_t readed;
-        do {
-            readed = read(fd1, buf1, BLOCK_SIZE);
-            if (memcmp(buf1, tmpFileContent, readed) != 0) {
-                lseek(fd1, 0, SEEK_SET);
-                auto writtenSize = write(fd1, newFileContent, newFileContentLen);
-                ret = (static_cast<unsigned long>(writtenSize) == newFileContentLen);
-                break;
-            }
-            tmpFileContent = tmpFileContent + BLOCK_SIZE;
-        } while (readed == BLOCK_SIZE);
-    } while (false);
+        tmpFileContent = tmpFileContent + BLOCK_SIZE;
+    } while (readed == BLOCK_SIZE);
 
-    if (fd1 != -1)
-        close(fd1);
+    if (fd != -1)
+        close(fd);
     return ret;
```

Comment:
Or `return true;`. And then you can remove lines with `ret` variable from function.